app/models/message.rb

```diff
@@ -2,15 +2,13 @@ class Message < ApplicationRecord
   has_many :reports, :through => :logs
   has_many :logs
   validates_lengths_from_database
-  validates :value, :digest, :presence => true
 
   def to_s
     value
   end
 
-  def self.find_or_create(val)
-    digest = Digest::SHA1.hexdigest(val)
-    Message.find_by(:digest => digest) || Message.create(:value => val, :digest => digest)
+  def self.find_or_create(value)
+    Message.find_or_create_by(value: value)
```

Comment:
nit: the `Message` is not needed
```suggestion
    find_or_create_by(value: value)
```
Should we deprecate this method and use `find_or_create_by` directly?

Reply:
Breaks at least two plugins, so deprecation message + this. But problem is this will flood logs with a lot of warnings :-(